lib/storage-handler.js

```diff
@@ -253,6 +253,7 @@ exports.download = function(provider, req, res, container, file, cb) {
     reader.pipe(res);
 
     reader.on('error', function onReaderError(err) {
+      res.set('content-type', 'application/json');
```

Comment:
this error wasnt being caught before due to the test case, i believe this is also an issue on [master](https://github.com/strongloop/loopback-component-storage/blob/e9b6a0ee19af29f359425102feca35805f8a5501/test/upload-download.test.js#L350) (also returning the error in `image/jpeg`), but the test case doesnt cover it, even tho there are errors
```coffeescript
Error: expected "Content-Type" matching /json/, got "image/jpeg; charset=utf-8"
    at Test.assert (/Users/davidcheung/node-web/loopback-component-storage/node_modules/supertest/lib/test.js:196:25)
    at assert (/Users/davidcheung/node-web/loopback-component-storage/node_modules/supertest/lib/test.js:132:12)
    at /Users/davidcheung/node-web/loopback-component-storage/node_modules/supertest/lib/test.js:129:5
    at Test.Request.callback (/Users/davidcheung/node-web/loopback-component-storage/node_modules/supertest/node_modules/superagent/lib/node/index.js:746:30)
    at /Users/davidcheung/node-web/loopback-component-storage/node_modules/supertest/node_modules/superagent/lib/node/index.js:880:14
    at IncomingMessage.<anonymous> (/Users/davidcheung/node-web/loopback-component-storage/node_modules/supertest/node_modules/superagent/lib/node/parsers/image.js:8:7)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:973:12)
    at /Users/davidcheung/node-web/loopback-component-storage/node_modules/async-listener/glue.js:188:31
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickDomainCallback [as _tickCallback] (internal/process/next_tick.js:122:9)
```